flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/SinkCodeGenerator.scala

```diff
@@ -83,8 +86,10 @@ object SinkCodeGenerator {
             inputRowType,
             inputTerm,
             inputFieldMapping = Option(mapping))
+        val outputBaseRowType = new BaseRowTypeInfo(
+          getCompositeTypes(pojo).map(fromTypeInfoToLogicalType): _*)
```

Comment:
It is pojo now, don't need `getCompositeTypes`.
Just `RowType.of((0 until ct.getArity).map(ct.getTypeAt).toArray)`.